http/gateway.js

```diff
@@ -62,8 +57,9 @@ var HttpGateway = Extendable.extend({
 
                     if (response.statusCode >= 400) {
                         if (response.statusCode === 401) {
-                            dispatcher.emit('401-response-received');
+                            gateway.handle401(resolve, reject, method, path, data, headers);
```

Comment:
This will fall through to the reject below. Is that what we want? Seems like maybe it should return on this line.